skpro/regression/gradient_boosting/_ngboost.py

```diff
@@ -0,0 +1,337 @@
+        minibatch_frac    : the percent subsample of rows to
+                            use in each boosting iteration
+        verbose           : flag indicating whether output
+                            should be printed during fitting
+        verbose_eval      : increment (in boosting iterations) at
+                            which output should be printed
+        tol               : numerical tolerance to be used in optimization
+        random_state      : seed for reproducibility.
+        validation_fraction: Proportion of training data to
+                             set aside as validation data for early stopping.
+        early_stopping_rounds: The number of consecutive
+            boosting iterations during which the
+            loss has to increase before the algorithm stops early.
+            Set to None to disable early stopping and validation
+            None enables running over the full data set.
+
+
+    Returns
+    -------
+        An NGBRegressor object that can be fit.
+    """
+
+    _tags = {
+        "python_dependencies": "ngboost",
+    }
+
+    from ngboost.distns import Laplace, LogNormal, Normal, T
+    from ngboost.learners import default_tree_learner
+    from ngboost.scores import LogScore
+
+    def __init__(
+        self,
+        dist="Normal",
+        score=LogScore,
+        base=default_tree_learner,
+        natural_gradient=True,
+        n_estimators=500,
+        learning_rate=0.01,
+        minibatch_frac=1.0,
+        col_sample=1.0,
+        verbose=True,
+        verbose_eval=100,
+        tol=1e-4,
+        random_state=None,
+        validation_fraction=0.1,
+        early_stopping_rounds=None,
+    ):
+        self.dist = dist
+        self.score = score
+        self.base = base
+        self.natural_gradient = natural_gradient
+        self.n_estimators = n_estimators
+        self.learning_rate = learning_rate
+        self.minibatch_frac = minibatch_frac
+        self.col_sample = col_sample
+        self.verbose = verbose
+        self.verbose_eval = verbose_eval
+        self.init_params = None
+        self.n_features = None
+        self.base_models = []
+        self.scalings = []
+        self.col_idxs = []
+        self.tol = tol
+        self.random_state = check_random_state(random_state)
+        self.best_val_loss_itr = None
+        self.validation_fraction = validation_fraction
+        self.early_stopping_rounds = early_stopping_rounds
+
+        super().__init__()
+
+    def dist_to_ngboost_instance(self):
+        """
+        Convert string to NGBoost object.
+
+        self.dist the input string for the type of Distribution.
+        It then creates an object of that particular NGBoost Distribution.
+
+        Returns
+        -------
+        NGBoost Distribution object.
+        """
+        from ngboost.distns import Laplace, LogNormal, Normal, Poisson, T
+
+        dist_ngboost = Normal
+        if self.dist == "Normal":
```

Comment:
yes, very nice, that's what I meant.

I would recommend to use a `dict` here, that makes the logic easier to maintain and extend.
Let me know if you want pointers how ot concretely use a `dict`.